src/libpmemobj/palloc.c

```diff
@@ -519,7 +519,7 @@ palloc_exec_actions(struct palloc_heap *heap,
 	/* perform all persistent memory operations */
 	operation_finish(ctx);
 
-	for (size_t i = 0; i < actvcnt; ++i) {
+	for (ssize_t i = (ssize_t)actvcnt - 1; i >= 0; --i) {
```

Comment:
this is scary (the idea of reversal, not implementation)

can we do something like that instead?
```
    if (i == actvcnt - 1 || act->lock != actv[i + 1].lock) {
```

Reply:
Done.